### ISSUE TITLE
RegionInfo.CurrentRegion.ISOCurrencySymbol returns XDR on iOS device

### ISSUE THREAD

**Issue by chamons** [dotnet/runtime#73464]
_From @Eilon on Thu, 04 Aug 2022 18:16:47 GMT_

---
**Issue moved from dotnet/maui#9182**
- Please respond to @softlion.

---

_From @softlion on Thursday, August 4, 2022 8:37:59 AM_

### Description

On an iOS device set to french FR/fr, RegionInfo.CurrentRegion.ISOCurrencySymbol returns XDR instead of EUR.  
This happens only on the device, not on the simulator.

Also, when using standard resource strings on a device with FR language set, the culture stays in EN as if FR was not detected.

Device: iPhone XR, iOS 15.5

![image](https://user-images.githubusercontent.com/190756/182781197-8533d8f3-a42f-410a-a3de-2bb3128b5bd5.png)

```
dotnet --version
6.0.400-preview.22330.6
VS 2022 Mac Preview 17.3.0.2089
with Xamarin.iOS 15.12.0.2
```

### Steps to Reproduce

* Set the iPhone langage/region to FR (france)
* Start the maiu app in debug mode
* Observe the value of RegionInfo.CurrentRegion.ISOCurrencySymbol

### Version with bug

Unknown/Other (please specify)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 15.5 device

### Did you find any workaround?

no

_Copied from original issue xamarin/xamarin-macios#15633_

**Comment by chamons** [comment]
_From @softlion on Thu, 04 Aug 2022 19:31:22 GMT_

I'm using MAUI iOS, not Xamarin @Eilon 
Why did you moved that here ?*

It's working perfectly fine on Xamarin. It's a migrated app.

**Comment by chamons** [comment]
_From @GalaxiaGuy on Fri, 05 Aug 2022 13:13:24 GMT_

The code for both Xamarin.iOS and .NET iOS exists in this repo.

**Comment by chamons** [comment]
_From @chamons on Fri, 05 Aug 2022 15:18:21 GMT_

So `RegionInfo.CurrentRegion.ISOCurrencySymbol` is a BCL API ([System.Globalization](https://docs.microsoft.com/en-us/dotnet/api/system.globalization?view=net-6.0)) and not part of macios. 

I believe https://developer.apple.com/documentation/foundation/nslocale/1642836-currencycode?language=objc would be the standard way of looking this up on iOS.

**Comment by chamons** [comment]
_From @chamons on Fri, 05 Aug 2022 15:20:11 GMT_

I am going to move this issue to dotnet/runtime as they own the BCL API in question.

**Comment by tarekgh** [comment]
@steveisok could you please triage this one? Thanks!

**Comment by steveisok** [comment]
Similar to https://github.com/dotnet/runtime/issues/70028.  This is something we will address in .NET 8.

**Comment by softlion** [comment]
Workaround. I still don't understand why it needs 2 releases ...  
It was working in Xamarin.

```
        public static string LocaleCurrency =>
#if IOS
            Foundation.NSLocale.CurrentLocale.CurrencyCode;
#elif ANDROID
            Java.Util.Currency.GetInstance(Android.App.Application.Context.Resources?.Configuration?.Locales.Get(0)!)?.CurrencyCode
            ?? RegionInfo.CurrentRegion.ISOCurrencySymbol;
#else
            RegionInfo.CurrentRegion.ISOCurrencySymbol;
#endif
```

**Comment by akoeplinger** [comment]
This is the same issue as mentioned in https://github.com/dotnet/runtime/issues/73512#issuecomment-1213464568

It was fixed in the 6.0.6 runtime release, I verified the issue doesn't happen with this or a later version.
